plugins/ai/copilot-chat.nix

```diff
@@ -0,0 +1,316 @@
+          end
+        ''
+        ''
+          Default selection (visual or line).
+          `fun(source: CopilotChat.config.source):CopilotChat.config.selection`
+        '';
+
+    prompts =
+      let
+        promptType = types.submodule {
+          options = {
+            prompt = helpers.mkNullOrStr "Prompt text.";
+
+            description = helpers.mkNullOrStr "Description for this prompt.";
+
+            kind = helpers.mkNullOrStr "Kind of this prompt.";
+
+            mapping = helpers.mkNullOrStr "A key to bind to this prompt.";
+
+            system_prompt = helpers.mkNullOrStr "System prompt to use.";
+
+            callback = helpers.mkNullOrLuaFn ''
+              Callback to trigger when this prompt is executed.
+              `fun(response: string, source: CopilotChat.config.source)`
+            '';
+
+            selection = helpers.mkNullOrLuaFn ''
+              Selection for this prompt.
+              `fun(source: CopilotChat.config.source):CopilotChat.config.selection`
+            '';
+          };
+        };
+
+        default = {
+          Explain.prompt = "/COPILOT_EXPLAIN Write an explanation for the active selection as paragraphs of text.";
+          Review = {
+            prompt = "/COPILOT_REVIEW Review the selected code.";
+            callback = ''
+              function(response, source)
+                -- see config.lua for implementation
+              end
+            '';
+          };
+          Fix.prompt = "/COPILOT_GENERATE There is a problem in this code. Rewrite the code to show it with the bug fixed.";
+          Optimize.prompt = "/COPILOT_GENERATE Optimize the selected code to improve performance and readablilty.";
+          Docs.prompt = "/COPILOT_GENERATE Please add documentation comment for the selection.";
+          Tests.prompt = "/COPILOT_GENERATE Please generate tests for my code.";
+          FixDiagnostic = {
+            prompt = "Please assist with the following diagnostic issue in file:";
+            selection = "require('CopilotChat.select').diagnostics";
+          };
+          Commit = {
+            prompt = "Write commit message for the change with commitizen convention. Make sure the title has maximum 50 characters and message is wrapped at 72 characters. Wrap the whole message in code block with language gitcommit.";
+            selection = "require('CopilotChat.select').gitdiff";
+          };
+          CommitStaged = {
+            prompt = "Write commit message for the change with commitizen convention. Make sure the title has maximum 50 characters and message is wrapped at 72 characters. Wrap the whole message in code block with language gitcommit.";
+            selection = ''
+              function(source)
+                return select.gitdiff(source, true)
+              end
+            '';
+          };
+        };
+      in
+      helpers.defaultNullOpts.mkAttrsOf (with types; either str promptType) default ''
```

Comment:
Should this be `attrsOf (maybeRaw (either str promptType))`?

Reply:
By default, `helpers.defaultNullOpts.mkAttrsOf` set the type to be `attrsOf (maybeRaw <TYPE>)`:
https://github.com/nix-community/nixvim/blob/d15fade62b743839a20d927d3506d503858f49f0/lib/options.nix#L122

Is that what you were talking about ?